qkit/drivers/Keysight_VNA_E5071C.py

```diff
@@ -916,31 +839,44 @@ def do_set_sweep_type(self,swtype):
         if swtype in ('LIN','LOG','SEGM','POW'):
             
             logging.debug(__name__ + ' : Setting sweep type to %s'%(swtype))
-            return self._visainstrument.write('SENS%i:SWE:TYPE %s' %(self._ci,swtype))
+            return self.write('SENS%i:SWE:TYPE %s' %(self._ci,swtype))
             
         else:
             logging.debug(__name__ + ' : Illegal argument %s'%(swtype))
-          
+    
     def write(self,msg):
-      return self._visainstrument.write(msg)    
-    def ask(self,msg):
-      return self._visainstrument.query(msg)
+        return self._visainstrument.write(msg)
+    
+    if qkit.visa.qkit_visa_version == 1:
+        def ask(self, msg):
+            return self._visainstrument.ask(msg)
+    
+        def ask_for_values(self, msg, **kwargs):
+            return self._visainstrument.ask_for_values(kwargs)
+    else:
+        def ask(self, msg):
+            return self._visainstrument.query(msg)
+    
+        def ask_for_values(self, msg, format=None, fmt=None):
+            dtype = format if format is not None else fmt if fmt is not None else qkit.visa.single
+            dtype = qkit.visa.dtypes[dtype]
+            return self._visainstrument.query_binary_values(msg,datatype=dtype,container=numpy.array)
      
     def pre_measurement(self):
         '''
         Set everything needed for the measurement
         '''
-        self._visainstrument.write(":TRIG:SOUR BUS")#Only wait for software triggers
-        self._visainstrument.write(":TRIG:AVER ON")# Tell the instrument to do the specific number of averages on every trigger.
+        self.write(":TRIG:SOUR BUS")#Only wait for software triggers
+        self.write(":TRIG:AVER ON")# Tell the instrument to do the specific number of averages on every trigger.
```

Comment:
If I remember correctly, @MartinSpiecker had some problems with that command. Their (new) VNAs were ignoring the trigger afterwards.

Reply:
If you have issues, please make a new issue or create a fix.